runner_http.go

```diff
@@ -51,8 +52,20 @@ func (r *runnerHTTP) Run(ctx context.Context, logger *slog.Logger, h Handler) {
 				}
 				return
 			}
-			f.Contents = nil
-			resp.Body = f
+			respBody := struct {
+				ContentType string `json:"content_type"`
+				Encoding    string `json:"encoding"`
+				Filename    string `json:"filename"`
+				SHA256      string `json:"sha256"`
+				Size        int    `json:"size,string"`
+			}{
+				ContentType: f.ContentType,
+				Encoding:    f.Encoding,
+				Filename:    f.Filename,
+				SHA256:      sha256Hash,
```

Comment:
the sha and size will be left to the runner to determine. This removes the chicken and egg problem where you need hte size and sha but want to work with the stream only. This isn't possible without having the runner do it. We can maintain streaming semantics while also obtaining our sha/size by extending the writer to support this when we're moving the contents to disk (or w/e sync we use)